File: standard-udfs-spark/src/test/scala/com/linkedin/stdudfs/spark/examples/TestMapValuesFunctionWrapper.scala
/**
 * BSD 2-CLAUSE LICENSE
 *
 * Copyright 2018 LinkedIn Corporation.
 * All Rights Reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are
 * met:
 *
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 *
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the
 *    distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
 * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
 * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
 * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
package com.linkedin.stdudfs.spark.examples

import com.linkedin.stdudfs.spark.common.AssertSparkExpression._
import org.testng.annotations.{BeforeClass, Test}

import scala.collection.mutable.WrappedArray._

class TestMapValuesFunctionWrapper {

  @BeforeClass
  def registerFunction(): Unit = {
    registerStandardUdf(
      "std_map_values",
      classOf[MapValuesFunctionWrapper]
    )
  }

  @Test
  def testMapValues(): Unit = {
    assertFunction("std_map_values(map(1, 4, 2, 5, 3, 6))", make(Array(4, 5, 6))) // scalastyle:ignore magic.number
    assertFunction("std_map_values(map('1', '4', '2', '5', '3', '6'))", make(Array("4", "5", "6")))
    assertFunction("std_map_values(null)", null)
  }
}

File: standard-udfs-avro/src/test/java/com/linkedin/stdudfs/avro/examples/TestMapValuesFunctionWrapper.java
package com.linkedin.stdudfs.avro.examples;

import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableMap;
import org.apache.avro.Schema;
import org.testng.annotations.Test;

import static com.linkedin.stdudfs.avro.common.AssertAvroUdf.*;
import static com.linkedin.stdudfs.avro.common.SchemaFactory.*;


public class TestMapValuesFunctionWrapper {
  @Test
  public void testMapValues() {
    assertFunction(new MapValuesFunctionWrapper(), new Schema[]{map(STRING, INTEGER)},
        new Object[]{ImmutableMap.of("1", 4, "2", 5, "3", 6)}, ImmutableList.of(4, 5, 6));

    assertFunction(new MapValuesFunctionWrapper(), new Schema[]{map(STRING, STRING)},
        new Object[]{ImmutableMap.of("1", "4", "2", "5", "3", "6")}, ImmutableList.of("4", "5", "6"));

    assertFunction(new MapValuesFunctionWrapper(), new Schema[]{map(STRING, STRING)}, new Object[]{null}, null);
  }
}
